server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1109,6 +1109,7 @@ protected Node(
                 clusterService.getClusterSettings(),
                 threadPool
             );
+            clusterService.setTaskResourceTrackingService(taskResourceTrackingService);
```

Comment:
Should we just initialize `TaskResourceTrackingService` in `ClusterService` then as we're creating a dependency here anyways.

Also should `SetOnce` be used?